### PR TITLE
Add initial CoreManager and CreateThreadWithClass

### PR DESCRIPTION
**Pull request by hq6**
This PR combines several closely related changes.
     
   - Introduce CoreManager as an interface.
   - Implement DefaultCoreManager and related tests as an implementation which
     supports normal and exclusive threads.
   - Remove virtualCoreTable from Arachne.

### CODE REVIEW

File: cwrapper/arachne_wrapper.cc [hunk 1]
```diff
@@ -60,6 +60,23 @@ arachne_wait_termination() {
     Arachne::waitForTermination();
 }
 
+/**
+ * This function is the wrapper for Arachne::createThreadWithClass.
```
Comment (qianl15) [inline cwrapper/arachne_wrapper.cc]: Probably add documentation about which class is the default thread class and which class is the exclusive one. Or maybe add an enum struct.

File: src/CoreManager.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+#include <string.h>
+#include <atomic>
+
+namespace Arachne {
+/*
+ * An unordered list of cores. This class's thread-safety is dependent on the
+ * thread-safety of torn reads.
+ */
+struct CoreList {
```
Comment (qianl15) [inline src/CoreManager.h]: May need to add unit tests for CoreList class.

File: cwrapper/arachne_wrapper.cc (hunk 1 above)
Comment (hq6) [reply cwrapper/arachne_wrapper.cc]: I've added documentation in the latest commit.

File: src/CoreManager.h (hunk 2 above)
Comment (hq6) [reply src/CoreManager.h]: Added unit tests.

**Comment by qianl15** [comment]
Need to resolve conflicts, then we can merge.

**Comment by hq6** [comment]
Manually merged.

**Pull request closed without merging**